### DISCUSSION TITLE
Add a price adjuster for payment methods

### DISCUSSION THREAD

**Discussion by andris-sevcenko** [craftcms/commerce#2301] [Ideas]
> Created by: Paul Verheul ([paul@nilsenpaul.nl](paul@nilsenpaul.nl)) on 2016/02/25 14:03:10 +0000
Votes at time of UserVoice import: 3

------
Some of our customers choose to let their customers pay a small fee, depending on the payment method they choose. Would love it if this could be a setting for the individual payment methods. Even better: make it possible to let this be a percentage of the total order amount (so you can let customers pay your PayPal fee).

**Comment by alexstojda** [comment]
This would be awesome!

**Comment by westhouseit** [comment]
I can't find any more information on this. Does anyone know of a good adjuster to add or should we just create one for our needs?

**Comment by anthonyjharrison** [comment]
Adding another voice to the request. In Australia it is very common (I'd actually say it's commonplace) for companies to charge a transaction fee when purchasing by card in store and now (rightly or wrongly) this practice is moving more and more online. If you modify the total prior to payment to include a fee Commerce reports it as 'Overpaid' which of course isn't a solution. Looking at the documentation, custom adjusters will no longer be supported in Commerce 5 https://craftcms.com/knowledge-base/dynamically-customizing-line-item-prices . Adjusting the price when populating the line item (as outlined in the URL) doesn't fit as it needs to be on the order, depending on payment method as Andris mentions.

This request for a first party price adjuster would be perfect! Please consider @brandonkelly and team.

**Comment by westhouseit** [comment]
We created a fee adjuster to add a custom order fee for e.g., all orders below £100. In part it looks like this:
```
/*
     * Example fee: [
     *  'enabled' => true,
     *  'name' => '$1 booking fee',
     *  'description' => '$1 fee on all bookings',
     *  'amount' => 1,
     *  'conditions' => [],
     * ]
*/
```
```
$adjustment = new OrderAdjustment;
$adjustment->type = 'fee';
$adjustment->name = $fee['name'];
$adjustment->description = $fee['description'];
$adjustment->amount = $fee['amount'];
$adjustment->setOrder($order);
$adjustments[] = $adjustment;
```

It sounds like this would break in v5, which is bad news for our clients. I can understand the issue with bad values being sent to payment gateways, but this works fine for us. Some means of adding a transaction fee is essential as can be seen by many multinational companies adding transaction fees to purchases. Perhaps this can be done with a positive number in the Discount adjuster, but I think we tried this and it wasn't right for some reason.

**Reply by samueldraper** [reply]
@brandonkelly any update on this situation. The lack of support for custom adjusters in Commerce 5 will mean no upgrade path for the client from Craft 4 unless the first party adjuster can be provided as suggested above. Do you have any other suggestions in the short term?
